### PR TITLE
Refactor/add unit tests and refactor route generator

### PR DESCRIPTION
**Pull request by eb-so**

### CODE REVIEW

File: packages/stacked_generator/lib/src/generators/exceptions/invalid_generator_input_exception.dart [hunk 1]
```diff
@@ -10,3 +10,10 @@ class InvalidGeneratorInputException implements Exception {
   String toString() =>
       'InvalidGeneratorInputException: message: $message\n todo: $todo\n element: $element';
 }
+
+class ExceptionMessages {
+  static const RouteClassNameShouldnotBeNull = 'className should not be null';
```
Comment (ferrarafer) [inline packages/stacked_generator/lib/src/generators/exceptions/invalid_generator_input_exception.dart:15]: Typo
RouteClassNameShould**n**otBeNull -> RouteClassNameShould**N**otBeNull

File: packages/stacked_generator/lib/src/generators/router/router_class_generator.dart [hunk 2]
```diff
@@ -129,6 +131,8 @@ class RouterClassGenerator extends BaseGenerator {
 
     var routesMap = <String, RouteConfig>{};
     routerConfig.routes?.forEach((route) {
+      throwIf(route.className == null,
+          ExceptionMessages.RouteClassNameShouldnotBeNull);
```
Comment (ferrarafer) [inline packages/stacked_generator/lib/src/generators/router/router_class_generator.dart]: Typo
RouteClassNameShould**n**otBeNull -> RouteClassNameShould**N**otBeNull

File: packages/stacked_generator/lib/src/generators/router/router_class_generator.dart [hunk 3]
```diff
@@ -145,10 +149,11 @@ class RouterClassGenerator extends BaseGenerator {
 
     if (r.parameters?.isNotEmpty == true) {
```
Comment (ferrarafer) [inline packages/stacked_generator/lib/src/generators/router/router_class_generator.dart]: I never saw this shortcut, how does it work? I assume is the same as doing `r.parameters?.isNotEmpty ?? true`,  right?

**Review by ferrarafer — COMMENTED** [review]
LGTM, just typo fixes

File: packages/stacked_generator/lib/src/generators/router/router_class_generator.dart (hunk 3 above)
Comment (eb-so) [reply packages/stacked_generator/lib/src/generators/router/router_class_generator.dart]: Me neither, this is 15 months old code, but I guess it should read like this 
1- if isNotEmpty is null then `null==true` which is false
2- if isNotEmpty is not null then it should execute normally

equals to
`r.parameters?.isNotEmpty ?? false`

**Comment by FilledStacks** [comment]
@eb-so I'm merging this in!

**Comment by eb-so** [comment]
I didn't finish up the work but that's ok I can create a new PR for the rest

**Pull request merged**